rfc/rfc006-distributed-registry.md

```diff
@@ -1,378 +1,582 @@
-
-##### CA certificate private key loss
-In case the vendor loses its CA private key it MUST register a new vendor CA certificate. This can be done in two ways,
-depending on what was lost:
-
-1. If the vendor still has a valid signing certificate it COULD sign the update using that private key.
-2. Otherwise, when the signing certificate has expired or the private key is lost as well, the vendor MUST also provide
-new proof to authenticate ownership of the vendor. The new proof MUST be the first operation in the **patch** field.
-
-Example update:
+The DID document has the following basic requirements:
+
+- the `id` MUST be generated according to the method specified at the beginning of §3
+- at least 1 key MUST be present in the `verificationMethod` and `authentication`
+- all key references in `authentication` MUST refer to keys listed under `verificationMethods`
+
+Each key listed in `verificationMethod` MUST have an `id` equal to the DID followed by a `#` and the public key fingerprint according to [rfc7638](https://tools.ietf.org/html/rfc7638)
+Each key MUST be of type `JsonWebKey2020` according to §5.3.1 of the [did-core-spec](https://www.w3.org/TR/did-core/#key-types-and-formats)
+
+The `controller` field MAY be present. This RFC follows the [did-core-spec](https://www.w3.org/TR/did-core/#did-controller).
+
+The `assertion` field MAY be present. 
+Keys referenced from this field are used for signing Verifiable Credentials/Presentations and for signing JWTs in the OAuth flow.
+
+Example DID document:
+
 ```json
 {
-  "patch": [
-    { "op": "add", "path": "/prfs", "value": "some new proof" },
-    { "op": "replace", "path": "/crts", "value": ["CA certificate as base64 ASN.1"] }
-  ]
+  "@context": [ "https://www.w3.org/ns/did/v1" ],
+  "id": "did:nuts:123",
+  "controller": [
+    "did:nuts:123",
+    "did:nuts:7368245"
+  ],
+  "verificationMethod": [{
+    "id": "did:nuts:123#_TKzHv2jFIyvdTGF1Dsgwngfdg3SH6TpDv0Ta1aOEkw",
+    "controller": "did:nuts:123",
+    "type": "JsonWebKey2020",
+    "publicKeyJwk": {
+      "crv": "P-256",
+      "x": "38M1FDts7Oea7urmseiugGW7tWc3mLpJh6rKe7xINZ8",
+      "y": "nDQW6XZ7b_u2Sy9slofYLlG03sOEoug3I0aAPQ0exs4",
+      "kty": "EC"
+    }
+  }],
+  "authentication": ["did:nuts:123#_TKzHv2jFIyvdTGF1Dsgwngfdg3SH6TpDv0Ta1aOEkw"],
+  "service": []
 }
-``` 
+```
+
+#### 3.2.2 Read (Resolve)
+A Nuts DID can only be resolved locally. The concept of the Nuts registry is the state based upon all Create, Update and Delete operations received through the Nuts Network.
+Therefore, any DID document SHOULD already be present in local storage.
 
-##### CA certificate private key leakage
-In case the vendor leaks its CA private key (or it gets stolen), it MUST remove the associated certificate as soon as possible
-and register a new one.
-In the same update it MUST register new proof and CA certificate as in the key loss process.
-Example update:
+##### 3.2.2.1 Resolution Input Metadata
```

Comment:
Resolution is about when you perform the read operation. This is done internally in the node. Does this paragraph actually define that historic data needs to be kept?

Reply:
Question is, do we need that functionality? If not, we should leave it out altogether

Reply:
Yes. Resolution description is mandatory for a DID method specification.